CSharpFunctionalExtensions.Tests/MaybeTests/Extensions/ToResultTests.Task.cs

```diff
@@ -1,53 +1,75 @@
-﻿using System.Threading.Tasks;
-using FluentAssertions;
-using Xunit;
-
-namespace CSharpFunctionalExtensions.Tests.MaybeTests.Extensions
-{
-    public class ToResultTests_Task : MaybeTestBase
-    {
-        [Fact]
-        public async Task ToResult_Task_returns_failure_if_has_no_value()
-        {
-            Maybe<T> maybe = null;
-
-            var result = await maybe.AsTask().ToResult("Error");
-
-            result.IsSuccess.Should().BeFalse();
-            result.Error.Should().Be("Error");
-        }
-
-        [Fact]
-        public async Task ToResult_Task_returns_success_if_has_value()
-        {
-            var maybe = Maybe<T>.From(T.Value);
-
-            var result = await maybe.AsTask().ToResult("Error".AsTask());
-
-            result.IsSuccess.Should().BeTrue();
-            result.Value.Should().Be(T.Value);
-        }
-
-        [Fact]
-        public async Task ToResult_Task_returns_custom_failure_if_has_no_value()
-        {
-            Maybe<T> maybe = null;
-
-            var result = await maybe.AsTask().ToResult(E.Value);
-
-            result.IsSuccess.Should().BeFalse();
-            result.Error.Should().Be(E.Value);
-        }
-
-        [Fact]
-        public async Task ToResult_Task_custom_failure_returns_success_if_has_value()
-        {
-            var maybe = Maybe<T>.From(T.Value);
-
-            var result = await maybe.AsTask().ToResult(E.Value.AsTask());
```

Comment:
Please verify if the `E.Value.AsTask()` thing was expected. I found it odd, and changed it to just `E.Value`. I did similar change in other test files.
If the `AsTask()` conversion was expected, do let me know, I'll revert this part.

Reply:
Not really sure. Maybe a copy-paste from other tests where we test both task- and non-task overloads of the methods. Here, we don't have an overload that accepts a task as the second argument, so we can remove these tests:
- ToResult_Task_returns_success_if_has_value
- ToResult_Task_custom_failure_returns_success_if_has_value

Reply:
Alright, I noticed that for Task and ValueTask, there was one redundant test file for each of them - the "Left" file had exactly the same tests as the "non-left" one. As you said, `ToResult` does not have `(Value)Task` as a second argument.

I removed the redundant file.

I fixed `ToResult_Task_returns_success_if_has_value` since it is a valid test
`ToResult_Task_custom_failure_returns_success_if_has_value` was alright I think, left it as it is.